wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGrpcGeneratorTest.kt

```diff
@@ -0,0 +1,291 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.kotlin
+
+import com.squareup.wire.schema.RepoBuilder
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class KotlinGrpcGeneratorTest {
+  @Test fun requestResponse() {
+    val expected = """
+          |package routeguide
+          |
+          |import com.squareup.wire.Service
+          |import com.squareup.wire.WireRpc
+          |import kotlin.coroutines.CoroutineContext
+          |
+          |interface RouteGuide : Service {
+          |    @WireRpc(
+          |            path = "/routeguide.RouteGuide/GetFeature",
+          |            requestAdapter = "routeguide.Point#ADAPTER",
+          |            responseAdapter = "routeguide.Feature#ADAPTER"
+          |    )
+          |    suspend fun GetFeature(context: CoroutineContext, request: Point): Feature
+          |}
+          |""".trimMargin()
+
+    val repoBuilder = RepoBuilder()
+        .add("routeguide.proto", """
+          |package routeguide;
+          |
+          |service RouteGuide {
+          |  rpc GetFeature(Point) returns (Feature) {}
+          |}
+          |$pointMessage
+          |$featureMessage
+          |""".trimMargin())
+    assertEquals(expected, repoBuilder.generateGrpcKotlin("routeguide.RouteGuide"))
+  }
+
+  @Test fun noPackage() {
+    val expected = """
+          |import com.squareup.wire.Service
+          |import com.squareup.wire.WireRpc
+          |import kotlin.coroutines.CoroutineContext
+          |
+          |interface RouteGuide : Service {
+          |    @WireRpc(
+          |            path = "/RouteGuide/GetFeature",
+          |            requestAdapter = "Point#ADAPTER",
+          |            responseAdapter = "Feature#ADAPTER"
+          |    )
+          |    suspend fun GetFeature(context: CoroutineContext, request: Point): Feature
+          |}
+          |""".trimMargin()
+
+    val repoBuilder = RepoBuilder()
+        .add("routeguide.proto", """
+          |service RouteGuide {
+          |  rpc GetFeature(Point) returns (Feature) {}
+          |}
+          |$pointMessage
+          |$featureMessage
+          |""".trimMargin())
+    assertEquals(expected, repoBuilder.generateGrpcKotlin("RouteGuide"))
+  }
+
+  @Test fun multiDepthPackage() {
+    val expected = """
+          |package routeguide.grpc
+          |
+          |import com.squareup.wire.Service
+          |import com.squareup.wire.WireRpc
+          |import kotlin.coroutines.CoroutineContext
+          |
+          |interface RouteGuide : Service {
+          |    @WireRpc(
+          |            path = "/routeguide.grpc.RouteGuide/GetFeature",
```

Comment:
nit: the wrapping +8 here seems wrong. KotlinPoet bug?

Reply:
known issue